sympy/core/tests/test_power.py

```diff
@@ -467,14 +468,28 @@ def test_issue_2993():
     assert str((-2.3*x - 2)**0.3) == '1.28386201800527*(-x - 0.869565217391304)**0.3'
     assert str((-2.3*x + 2)**0.3) == '1.28386201800527*(0.869565217391304 - x)**0.3'
     assert str((2.3*x + 2)**0.3) == '1.28386201800527*(x + 0.869565217391304)**0.3'
-    assert str((2.3*x - 4)**Rational(1, 3)) == '1.5874010519682*(0.575*x - 1)**(1/3)'
+    assert str((2.3*x - 4)**Rational(1, 3)) == '2**(2/3)*(0.575*x - 1)**(1/3)'
     eq = (2.3*x + 4)
-    assert eq**2 == 16.0*(0.575*x + 1)**2
+    assert eq**2 == 16*(0.575*x + 1)**2
     assert (1/eq).args == (eq, -1)  # don't change trivial power
     # issue 17735
     q=.5*exp(x) - .5*exp(-x) + 0.1
     assert int((q**2).subs(x, 1)) == 1
-
+    # issue 17756
+    y = Symbol('y')
+    assert len(sqrt(x/(x + y)**2 + Float('0.008', 30)).subs(y, pi.n(25)).atoms(Float)) == 2
+    r_at_max_density,x1,x8,x10,x11,x12 = symbols('r_at_max_density x1 x8 x10 x11 x12',real=True)
+    expr = sqrt(1 + N('0.0170992456333788667034850458615',30)*(r_at_max_density**4 + x1*x10 - 2*x1*x11 - x12*(-x1 + x10))**2/(r_at_max_density**3*x8**2*(x10 - 3*x11 + 2*x12)**2))/2
+    expr = expr.subs(x8, N('0.0966594956075474769169134801223',30))
+    ## ^ above subs() is where issue #17756 loss of significance occurs.
```

Comment:
I actually don't see the difference between this PR and master at this step. Suggest:
```python
# issue 17756
a, b, c, d, e, f = symbols('a:f')
expr = sqrt(1 + a*(c**4 + g*d - 2*g*e - f*(-g + d))**2/
    (c**3*b**2*(d - 3*e + 2*f)**2))/2
r = [
(a, N('0.0170992456333788667034850458615', 30)),
(b, N('0.0966594956075474769169134801223', 30)),
(c, N('0.390911862903463913632151616184', 30)),
(d, N('0.152812084558656566271750185933', 30)),
(e, N('0.137562344465103337106561623432', 30)),
(f, N('0.174259178881496659302933610355', 30)),
(g, N('0.220745448491223779615401870086', 30))]
tru = expr.n(30, subs=dict(r))
seq = expr.subs(r)
# although `tru` is the right way to evaluate
# expr with numerical values, `seq` will have
# significant loss of precision if extraction of
# the largest coefficient of a power's base's terms
# is done improperly
assert seq == tru
```

Reply:
I like it. Much cleaner, and does the job well (assertion fails in master but passes beautifully after this PR). Committed.

Reply:
If you decide to clean up the history with local commands
```
git diff 78d5a35 > dif
git co 78d5a35
git branch -b tmp
git apply dif
```
Then you could modify the line were addpow is calculated to
```
addpow = Add(*[c*m for c, m in zip(c, m)])**e
```
(no need to rename c to d). And then commit with a single commit with a short title and push it here with the `force` option.
```
git commit -am 'your short title'
git branch -D master
git co -b master
git branch -D tmp
git push -f zachetienne master
```

Reply:
@smichr : Okay, I followed your instructions as best I could (co == checkout, etc). I think I did the correct thing. Please take a look and let me know. Thanks!

Reply:
Oops (re `co`). And yes, it looks good. But if you reedit the commit message and put a blank line after the title then you won't get the whole commit message as the title.

```
Fix issue #17756

PR #16762 causes a reduction of precision to 16 significant digits. 

Special thanks to Christopher Smith (@smichr) and Isuru Fernando (@isuruf)
for their valuable input and contributions to this patch.
```

Reply:
Okay, I think that (git commit --amend) did it.